zkevm-circuits/src/evm_circuit/execution/balance.rs

```diff
@@ -0,0 +1,300 @@
+
+#[derive(Clone, Debug)]
+pub(crate) struct BalanceGadget<F> {
+    same_context: SameContextGadget<F>,
+    address: RandomLinearCombination<F, N_BYTES_ACCOUNT_ADDRESS>,
+    reversion_info: ReversionInfo<F>,
+    tx_id: Cell<F>,
+    is_warm: Cell<F>,
+    balance: Cell<F>,
+    exists: Cell<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for BalanceGadget<F> {
+    const NAME: &'static str = "BALANCE";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::BALANCE;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let address = cb.query_rlc();
+        cb.stack_pop(address.expr());
+
+        let tx_id = cb.call_context(None, CallContextFieldTag::TxId);
+        let mut reversion_info = cb.reversion_info_read(None);
+        let is_warm = cb.query_bool();
+        cb.account_access_list_write(
+            tx_id.expr(),
+            from_bytes::expr(&address.cells),
+            1.expr(),
+            is_warm.expr(),
+            Some(&mut reversion_info),
+        );
+
+        let exists = cb.query_bool();
+        let balance = cb.condition(exists.expr(), |cb| {
+            let balance = cb.query_cell();
+            cb.account_read(
+                from_bytes::expr(&address.cells),
+                AccountFieldTag::Balance,
+                balance.expr(),
+            );
+            balance
+        });
+        cb.condition(1.expr() - exists.expr(), |cb| {
+            cb.account_read(
+                from_bytes::expr(&address.cells),
+                AccountFieldTag::NonExisting,
+                0.expr(),
+            );
+        });
+
+        cb.stack_push(select::expr(exists.expr(), balance.expr(), 0.expr()));
+
+        let gas_cost = is_warm.expr() * GasCost::WARM_ACCESS.expr()
+            + (1.expr() - is_warm.expr()) * GasCost::COLD_ACCOUNT_ACCESS.expr();
+
+        let step_state_transition = StepStateTransition {
+            rw_counter: Delta(cb.rw_counter_offset()),
+            program_counter: Delta(1.expr()),
+            stack_pointer: Delta(0.expr()),
+            gas_left: Delta(-gas_cost),
+            reversible_write_counter: Delta(1.expr()),
+            ..Default::default()
+        };
+
+        let opcode = cb.query_cell();
+        let same_context = SameContextGadget::construct(cb, opcode, step_state_transition);
+
+        Self {
+            same_context,
+            address,
+            reversion_info,
+            tx_id,
+            is_warm,
+            balance,
+            exists,
+        }
+    }
+
+    fn assign_exec_step(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        block: &Block<F>,
+        tx: &Transaction,
+        call: &Call,
+        step: &ExecStep,
+    ) -> Result<(), Error> {
+        self.same_context.assign_exec_step(region, offset, step)?;
+
+        let address = block.rws[step.rw_indices[0]].stack_value().to_address();
+        let mut address_bytes = address.0;
+        address_bytes.reverse();
```

Comment:
want to know what reason for `reverse` it

Reply:
Try to investigate `ethers_core::types::Address` and code in `eth-types`, it seems that the `Address` is big endian.

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/eth-types/src/lib.rs#L167